src/LowHash0.cpp

```diff
@@ -356,6 +367,44 @@ void LowHash0::pass1ThreadFunction(size_t threadId)
 
 }
 
+size_t LowHash0::computeMinBucketSizeIfNotProvided() {
+    size_t minBucketSize = 0;
+    vector<uint64_t> histogram(bucketHistogram.size(), 0);
+    
+    for (uint64_t bucketSize = 0; bucketSize < bucketHistogram.size(); bucketSize++) {
+        const auto frequency = bucketHistogram[bucketSize];
+        histogram[bucketSize] = bucketSize * frequency;
+    }
+
+    try {
+        PeakFinder p;
+        p.findPeaks(histogram);
+        minBucketSize = p.findXCutoff(histogram);
+        cout << timestamp << "MinHash.minBucketSize = " << minBucketSize << " (according to PeakFinder)" << endl;
+    } catch (const PeakFinderException& e) {
+        uint64_t prevFeatureCount = 0;
+        for (uint64_t bucketSize = 0; bucketSize < histogram.size(); bucketSize++) {
+            if (prevFeatureCount == 0) {
+                prevFeatureCount = histogram[bucketSize];
+                continue;
+            }
+
+            double ratio = double(histogram[bucketSize]) / double(prevFeatureCount);
+            if (ratio > 0.8) {
```

Comment:
Do you mean to check for `ratio < 0.8`?

Reply:
No. I want the drop to be less than 20% - the reduced slope indicating that we're close to bottoming out. May be a more readable version of this would be ...
```
double drop = (prevFeatureCount - histogram[bucketSize])/prevFeatureCount;
if (drop <= 0.2) {
  ...
}
```

Reply:
I see. Honestly, I am not sure how robust this will be. I am not sure I want to offer this to users as an alternative for dealing with setting the value themselves. Even if we do it with a big disclaimer, people could start using it indiscriminately.